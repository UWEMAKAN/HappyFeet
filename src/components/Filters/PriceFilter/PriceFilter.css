.PriceFilter select {
  width: 6rem;
  font-size: 1.4rem;
  outline: none;
  border: 1px solid rgba(0,0,0,.4);

  padding: .3rem;
  background-color: #fff;
}

.PriceFilter span {
  margin: 0 .5rem;
}

@media screen and (max-width: 900px) {
  .PriceFilter select {
    font-size: 1.2rem;
    outline: none;
    border: 1px solid rgba(0,0,0,.4);

    padding: .2rem;
    background-color: #fff;
  }
}
