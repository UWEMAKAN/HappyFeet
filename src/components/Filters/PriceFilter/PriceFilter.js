import React from 'react';
import propTypes from 'prop-types';
import './PriceFilter.css';

const PriceFilter = ({ selectHandler }) => (
  <div className="Filter PriceFilter">
    <h3>Price</h3>
    <div>
      <select onChange={selectHandler} name="minPriceSelector">
        <option value={null}>Min</option>
        <option value={0}>0</option>
        <option value={500}>500</option>
        <option value={1000}>1000</option>
        <option value={1500}>1500</option>
        <option value={2000}>2000</option>
        <option value={2500}>2500</option>
        <option value={3000}>3000</option>
        <option value={3500}>3500</option>
        <option value={4000}>4000</option>
      </select>
      <span>to</span>
      <select onChange={selectHandler} name="maxPriceSelector">
        <option value={null}>Max</option>
        <option value={500}>500</option>
        <option value={1000}>1000</option>
        <option value={1500}>1500</option>
        <option value={2000}>2000</option>
        <option value={2500}>2500</option>
        <option value={3000}>3000</option>
        <option value={3500}>3500</option>
        <option value={4000}>4000</option>
        <option value={4001}>4000+</option>
      </select>

    </div>
  </div>
);

PriceFilter.propTypes = {
  selectHandler: propTypes.func.isRequired
};

export default PriceFilter;
