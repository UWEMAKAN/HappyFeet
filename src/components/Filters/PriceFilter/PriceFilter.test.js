import React from 'react';
import { shallow } from 'enzyme';
import PriceFilter from './PriceFilter';

const renderPriceFilter = (args) => {
  const defaultProps = {
    selectHandler: jest.fn()
  };
  const props = { ...defaultProps, ...args };
  return shallow(<PriceFilter selectHandler={props.selectHandler} />);
};

describe('PriceFilter Component', () => {
  test('should render PriceFilter Component', () => {
    const selectHandler = jest.fn();
    const wrapper = renderPriceFilter({ selectHandler });
    expect.assertions(3);
    expect(wrapper.find('h3').text()).toEqual('Price');
    expect(wrapper.getElements()[0].props.children[1].props.children.length).toBe(3);
    expect(wrapper).toMatchSnapshot();
  });
});
