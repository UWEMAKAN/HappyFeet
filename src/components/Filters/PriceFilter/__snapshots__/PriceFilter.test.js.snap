// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceFilter Component should render PriceFilter Component 1`] = `ShallowWrapper {}`;
